pkg/grpc/api.go

```diff
@@ -494,3 +509,8 @@ func (a *api) ExecuteStateTransaction(ctx context.Context, in *runtimev1pb.Execu
 	}
 	return &empty.Empty{}, nil
 }
+
+func (a *api) isSecretAllowed(storeName, key string) bool {
+	return config.IsSecretAllowed(storeName, key, a.defaultSecretAccess[storeName],
+		a.allowedSecrets, a.deniedSecrets)
```

Comment:
Can we please either perform all the map lookups here or in `IsSecretAllowed()`? Currently we look up for `a.defaultSecretAccess[storeName]` here but  `a.allowedSecrets[storeName]` in the `config.IsSecretAllowed`.

If we do all the lookup here we could drop `storeName` in the signature.